backend/app/api/auth.py

```
@@ -0,0 +1,151 @@
+        f"state={state}&"
+        f"scope=user:email"
+    )
+    
+    return {"auth_url": auth_url, "state": state}
+
+
+@router.post("/github/callback")
+async def github_callback(
+    code: str = Query(...),
+    state: str = Query(...),
+    db: AsyncSession = Depends(get_db),
```

Ruff (0.15.6)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)